app/grandchallenge/components/models.py

```diff
@@ -1504,7 +1504,7 @@ class ComponentJob(FieldChangeMixin, UUIDModel):
     runtime_metrics = models.JSONField(default=dict, editable=False)
     error_message = models.CharField(max_length=1024, default="")
     detailed_error_message = models.JSONField(
-        blank=True, null=True, default=None
+        blank=True, null=True, default=dict
```

Comment:
Changing this to an empty dictionary by default since that is how it is used everywhere in the codebase, and if it were ever used with a list or, things would break, so making this explicit here seems the way to go.

Reply:
Do the existing values need to be updated? Should we disallow `null` so that there are not 2 different representations of empty data?

Reply:
If the default is the only thing being changed then new migrations are not necessary. The default is only used when the field is added so the new migrations will not have any SQL changes. If this is the case then the new migrations can be deleted and the latest ones that touches these fields can be modified.

Reply:
Okay then that would require a new migration so no need to edit the old ones.